iree/compiler/Dialect/VM/Conversion/VMToEmitC/ConvertVMToEmitC.cpp

```diff
@@ -28,6 +30,46 @@ namespace iree_compiler {
 
 namespace {
 
+LogicalResult sanitizeSymbolNames(IREE::VM::ModuleOp &moduleOp) {
+  auto replaceIllegalChars = [](std::string &symbolName) {
+    std::regex illegalCharactersRe("<|>");
+    auto beginIt = std::sregex_iterator(symbolName.begin(), symbolName.end(),
+                                        illegalCharactersRe);
+    auto endIt = std::sregex_iterator();
+
+    // Replace illegal characters in symbol name with "_" using iterator
+    for (std::sregex_iterator i = beginIt; i != endIt; ++i)
+      symbolName.replace(i->position(), i->length(), "_");
+
+    // Returns true if there was a match
+    return beginIt != endIt;
+  };
+
+  std::string symbolName(moduleOp.getName());
+
+  // Sanitize name and save original name if needed
+  if (replaceIllegalChars(symbolName)) {
+    // Save original module name as an attribute
+    moduleOp.getOperation()->setAttr(
+        "vm.module_name",
+        StringAttr::get(moduleOp.getContext(), moduleOp.getName()));
+  }
+
+  SymbolTable::setSymbolName(moduleOp, symbolName);
+
+  for (auto op : moduleOp.getOps<SymbolOpInterface>()) {
+    std::string symbolName(op.getName());
+    replaceIllegalChars(symbolName);
```

Comment:
Just to confirm, the names used for exported function names are pulled from `vm.export`, correct? (that won't get changed here as it is not a symbol op and the internal names are fine to be differ)